### ISSUE TITLE
 {"error":"Invalid bucket"}

### ISSUE THREAD

**Issue by azad-techvilla** [vimeo/vimeo.php#203]
Getting Client error: `POST https://files.tus.vimeo.com/files/vimeo-prod-src-tus-asia` resulted in a `400 Bad Request` response: {"error":"Invalid bucket"}
while uploading video 

![Screenshot from 2019-03-15 12-21-06](https://user-images.githubusercontent.com/47494673/54413848-35939980-471d-11e9-9e55-41f6ff2b8b06.png)

Source:

     protected $lib;

    public function __construct()
    {
        $client_id='xxxx';
        $client_secret='xxxxx+xxxxxx+xxxx/xxxx';
        $access_token='xxxxx';
        $lib = new \Vimeo\Vimeo($client_id, $client_secret,$access_token);
       $this->lib = $lib;
    }

    function upload($source_path)
    {
        try {
        $response = $this->lib->upload('./../../publicdata/temp/video/test.mp4', [
            'name' => 'VP Test API Upload',
            'privacy' => [
                'view' => 'anybody'
            ]
        ]);
        echo $print_r($response);
        } catch (VimeoUploadException $e) {
            // We may have had an error. We can't resolve it here necessarily, so report it to the user.
            echo 'Error uploading ' . $file_name . "\n";
            echo 'Server reported: ' . $e->getMessage() . "\n";
        } catch (VimeoRequestException $e) {
            echo 'There was an error making the request.' . "\n";
            echo 'Server reported: ' . $e->getMessage() . "\n";
        }
    }

**Comment by tommypenner** [comment]
@azad-techvilla What version of the Vimeo PHP library do you have installed? This issue was previously reported and fixed by upgrading to version 3.0.2, take a look at this ticket for reference: https://github.com/vimeo/vimeo.php/issues/189#issuecomment-448272971

**Comment by WyattCast44** [comment]
Updating vimeo/vimeo-api to 3.0.2 worked for me as well. If using in Laravel, I had to require `vimeo/vimeo-api` before requiring `vimeo/laravel`.

**Comment by wangjacsi** [comment]
I got the same error

**Comment by lc-brito** [comment]
Can't update to 3.0.2 using Laravel 5.8, it conflicts with nesbot/carbon.

**Comment by adamanderson86** [comment]
Anyone had any luck with this. I'm running laravel 5.8 (downgraded from 6.0). Thanks

**Comment by mstaack** [comment]
any updates on this?

**Comment by philipnewcomer** [comment]
I'm experiencing this issue on (at the time of this writing) the latest release of vimeo/vimeo-api, 3.0.3.

**Comment by tommypenner** [comment]
No updates about this issue at the moment, but a few breadcrumbs we uncovered today:

- User reports of this issue show a `POST` request to the upload_link `https://files.tus.vimeo.com/files/vimeo-prod-src-tus-us` (or their regional tus upload server: asia, eu, etc.)
- The correct method is `PATCH`, not `POST` per [tus docs](https://developer.vimeo.com/api/upload/videos#resumable-approach-step-2).
- The correct upload_link includes an additional string at the end, like `https://files.tus.vimeo.com/files/vimeo-prod-src-tus-us/aaabbbcccddd000111222333`

While testing today, we verified that the initial request to `POST /me/videos` returns the correct upload_link from the server, so somewhere client-side between that response and the following file upload step is modifying the upload_link.

We'll be looking into this ASAP, especially now that we are able to reproduce the issue.
